website/events/views.py

```diff
@@ -179,3 +181,30 @@ def dispatch(self, request, *args, **kwargs):
         except RegistrationError:
             pass
         return redirect(self.event)
+
+
+@method_decorator(login_required, name="dispatch")
+class MarkPresentView(View):
+    """A view that allows uses to mark their presence at an event using a secret token."""
+
+    def get(self, request, *args, **kwargs):
+        """Mark a user as present.
+
+        Checks if the url is correct, the event has not ended yet, and the user is registered.
+        """
+        event = get_object_or_404(Event, pk=kwargs["pk"])
+        if kwargs["token"] != event.mark_present_url_token:
+            messages.error(request, _("Invalid url."))
+        elif not request.member or not is_user_registered(request.member, event):
+            messages.error(request, _("You are not registered for this event."))
+        elif event.end < timezone.now():
+            messages.error(request, _("This event has already ended."))
+        else:
```

Comment:
I think we should add a warning or info message if the user was already marked present